### DISCUSSION TITLE
Can we just plot pole figures of subset of the grains from EBSD measurements?

### DISCUSSION THREAD

**Discussion by yuehengzhang95** [mtex-toolbox/mtex#1363] [Ask Anything] [answered]
Hi, I would like to choose just a subset of the grains from EBSD measurements. Is there a way for me to just see the pole figures of such subset?

Here is the code for the creating the pole figures for the whole dataset.
`%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('321', [4.8 4.8 13], 'X||a', 'Y||b*', 'Z||c*', 'mineral', 'Alumina', 'color', [0.56 0.74 0.56]),...
  crystalSymmetry('432', [4.1 4.1 4.1], 'mineral', 'Gold', 'color', [0.53 0.81 0.98])};

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','intoPlane');

%% Specify File Names

% path to files
pname = '/Users/yuehengzhang/Downloads';

% which files to be imported
fname = [pname '/Scan_P6.ang'];

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertEuler2SpatialReferenceFrame','setting 2');

%%
% plotx2east % this command tells MTEX to plot the x coordinates increasing to the east (left)
% plot(ebsd,'coordinates','on')

% mtexdata small
[grains,ebsd('indexed').grainId] = calcGrains(ebsd('indexed'));
ebsd(grains(grains.grainSize<=10)) = [];
[grains,ebsd('indexed').grainId] = calcGrains(ebsd('indexed'));
F = splineFilter;
ebsd = smooth(ebsd('indexed'),F,'fill',grains);
[grains,ebsd('indexed').grainId] = calcGrains(ebsd('indexed'));

gold = grains('Gold')

odf = unimodalODF(gold.meanOrientation)

plotPDF(odf,Miller({1,0,0},{1,1,0},{1,1,1},gold.CS))
odf_rotated = rotate(odf, rotation('axis', yvector, 'angle', -3*degree));
odf_rotated_2 = rotate(odf_rotated, rotation('axis', zvector, 'angle', -2*degree));
odf_rotated_3 = rotate(odf_rotated_2, rotation('axis', xvector, 'angle', 2*degree));
plotPDF(odf_rotated_3,h,'figSize','medium')
mtexColorbar`

**Comment by kilir** [comment, marked as answer]
Hi,
have a look here: https://mtex-toolbox.github.io/SelectingGrains.html#11
Cheers,
Rüdiger
